common/context/src/test/java/io/helidon/common/context/ContextAwareScheduledExecutorTest.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.common.context;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Unit test for {@link ContextAwareExecutorImpl}.
+ */
+public class ContextAwareScheduledExecutorTest {
+    private static final String TEST_STRING = "someStringToTest";
+
+    @Test
+    void testSchedule() throws Exception {
+        ScheduledExecutorService wrapped = Executors.newSingleThreadScheduledExecutor();
+        ScheduledExecutorService wrapper = Contexts.wrap(wrapped);
+
+
+        CountDownLatch cdl = new CountDownLatch(1);
+        Context context = Context.create();
+        context.register(cdl);
+
+        AtomicReference<ScheduledFuture<?>> futureRef = new AtomicReference<>();
+        AtomicReference<Exception> exceptionRef = new AtomicReference<>();
+
+        Contexts.inContext(context, () -> {
+            futureRef.set(wrapper.schedule(() -> {
+                try {
+                    Contexts.context()
+                            .get()
+                            .get(CountDownLatch.class)
+                            .get()
+                            .countDown();
```

Comment:
Wow, that's kind of an ugly/confusing sequence at first sight; it is what we get with layered Optionals though.

Reply:
In general we should not have public API methods that return null. We should always return `Optional` instead.
This is ugly because it is a test that intentionally wants to fail. In real world, you should never expect something to be in the context and fail with a NPE (or no such element exception in this case) - you must do something about it. 
So `Optional` is intentional and correct.

Reply:
Also a general answer to all the comments - as I have decided to have the API backward compatible and the webserver context implements the common context, we need to keep these APIs.